persist/persist_test.go

```diff
@@ -26,3 +26,71 @@ func TestIntegrationRandomSuffix(t *testing.T) {
 		file.Close()
 	}
 }
+
+// TestUnitNewSafeFile checks that a new file is created and that its name is
+// different than the finalName of the new safeFile.
+func TestUnitNewSafeFile(t *testing.T) {
```

Comment:
technically not a unit test because you are using the actual os package, which is IO and slow.